### PR TITLE
Add processor serialization.

### PR DESCRIPTION
**Pull request by siarheishut**
Add processor serialization and deserialization methods.

### CODE REVIEW

File: src/request_processor.cpp [hunk 1]
```diff
@@ -106,6 +106,27 @@ json::Dict ToJson(MapResponse resp, int id) {
...
 
+std::unique_ptr<Processor> Processor::Deserialize(
+    const ::TransportCatalog::TransportCatalog &proto_catalog) {
+  auto catalog = std::shared_ptr(
+      rm::TransportCatalog::Deserialize(proto_catalog.database()));
+  auto route_manager = rm::RouteManager::Deserialize(
+      proto_catalog, std::shared_ptr<TransportCatalog>(catalog.get()));
+  auto route_manager_sptr = std::shared_ptr(std::move(route_manager));
+  auto bus_manager = std::make_unique<BusManager>(
+      std::shared_ptr<TransportCatalog>(catalog.get()),
```
Comment (foxfrikses) [inline src/request_processor.cpp:117]: What is transport catalog. Please, use full names with all the namespaces here. It seems you're getting segfault

Comment (siarheishut) [reply src/request_processor.cpp:117]: Done.

File: src/request_processor.cpp [hunk 2]
```diff
@@ -106,6 +106,26 @@ json::Dict ToJson(MapResponse resp, int id) {
   return json::Dict{{"request_id", id}, {"map", resp.map}};
 }
 
+std::unique_ptr<Processor> Processor::Deserialize(
+    const ::TransportCatalog::TransportCatalog &proto_catalog) {
+  auto catalog = std::shared_ptr(
+      rm::TransportCatalog::Deserialize(proto_catalog.database()));
+
+  auto route_manager = rm::RouteManager::Deserialize(proto_catalog, catalog);
+  auto route_manager_sptr = std::shared_ptr(std::move(route_manager));
```
Comment (foxfrikses) [inline src/request_processor.cpp:115]:
```suggestion
  auto route_manager = std::shared_ptr(
      rm::RouteManager::Deserialize(proto_catalog, catalog));
```

File: src/request_processor.cpp [hunk 3]
```diff
@@ -106,6 +106,26 @@ json::Dict ToJson(MapResponse resp, int id) {
...
+
+  auto bus_manager = std::make_unique<BusManager>(catalog, route_manager_sptr);
+  RenderingSettings rendering_settings;
+  rendering_settings.Deserialize(proto_catalog.rendering_settings());
+  auto map_renderer = rm::MapRenderer::Create(catalog, rendering_settings);
+  return std::unique_ptr<Processor>(
+      new Processor(std::move(catalog),
+                    std::move(route_manager_sptr),
+                    std::move(bus_manager),
+                    std::move(map_renderer),
```
Comment (foxfrikses) [inline src/request_processor.cpp:125]: let's add a null-check for this beauties

File: src/request_processor.cpp (hunk 2 above)
Comment (siarheishut) [reply src/request_processor.cpp:115]: Done.

File: src/request_processor.cpp (hunk 3 above)
Comment (siarheishut) [reply src/request_processor.cpp:125]: Done.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**